packages/camera/camera_android_camerax/lib/src/image_analysis.dart

```diff
@@ -0,0 +1,183 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:camera_platform_interface/camera_platform_interface.dart'
+    show CameraImageData, CameraImageFormat, CameraImagePlane, ImageFormatGroup;
+import 'package:flutter/services.dart' show BinaryMessenger;
+
+import 'android_camera_camerax_flutter_api_impls.dart';
+import 'camerax_library.g.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+import 'use_case.dart';
+
+/// Use case for providing CPU accessible images for performing image analysis.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/ImageAnalysis.
+class ImageAnalysis extends UseCase {
+  /// Creates an [ImageAnalysis].
+  ImageAnalysis(
+      {BinaryMessenger? binaryMessenger,
+      InstanceManager? instanceManager,
+      this.targetResolution})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = ImageAnalysisHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+    _api.createFromInstance(this, targetResolution);
+    AndroidCameraXCameraFlutterApis.instance.ensureSetUp();
+  }
+
+  /// Constructs an [ImageAnalysis] that is not automatically attached to a native object.
+  ImageAnalysis.detached(
+      {BinaryMessenger? binaryMessenger,
+      InstanceManager? instanceManager,
+      this.targetResolution})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = ImageAnalysisHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+    AndroidCameraXCameraFlutterApis.instance.ensureSetUp();
+  }
+
+  /// Stream that emits data whenever a frame is received for image streaming.
+  static final StreamController<CameraImageData>
+      onStreamedFrameAvailableStreamController =
+      StreamController<CameraImageData>.broadcast();
+
+  late final ImageAnalysisHostApiImpl _api;
+
+  /// Target resolution of the camera preview stream.
+  final ResolutionInfo? targetResolution;
+
+  /// Configures this instance for image streaming support.
+  ///
+  /// This is a direct wrapping of the setAnalyzer method in CameraX,
+  /// but also handles the creation of the CameraX ImageAnalysis.Analyzer
+  /// that is used to collect the image information required for image
+  /// streaming.
+  ///
+  /// See [ImageAnalysisFlutterApiImpl.onImageAnalyzed] for the image
+  /// information that is analyzed by the created ImageAnalysis.Analyzer
+  /// instance.
+  Future<void> setAnalyzer() async {
```

Comment:
The Java method takes an [ImageAnalysis.Analyzer](https://developer.android.com/reference/androidx/camera/core/ImageAnalysis.Analyzer). Why does this not take any parameters?

Reply:
Because it's not a direct wrapping. Like I said in the comment, it also handles the creation of that analyzer. Even if we make changes to what exactly is sent back to the Dart side (image info versus `ImageProxy`), I don't see why we would wrap `ImageAnalysis.Analyzer`.

Reply:
Ah, I hadn't realized you referenced in the docs. That's fine then.